### PR TITLE
Jenkinsfile

### PR DESCRIPTION
**Pull request by jgmize**
Example run with this branch right before the last commit, which switches stage to master: https://ci.us-west.moz.works/blue/organizations/jenkins/irlpodcast/detail/jenkinsfile/6/pipeline/16
Staging site built and pushed with the above job: https://stage.irlpodcast.org/

### CODE REVIEW

**Comment by jgmize** [comment]
cc @metadave

**Comment by metadave** [comment]
lgtm, :shipit:

File: bin/irc-notify.sh [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+set -eo pipefail
+
+# Required environment variables if using --stage:
+# BRANCH_NAME, BUILD_NUMBER
+
+# defaults and constants
+NICK="irlbot"
+CHANNEL="#www"
+SERVER="irc.mozilla.org:6697"
+BLUE_BUILD_URL="https://ci.us-west.moz.works/blue/organizations/jenkins/irlpodcast"
```
Comment (jgmize) [inline bin/irc-notify.sh]: I copied this file from https://github.com/mozmar/basket/blob/master/bin/irc-notify.sh and modified the vars above.

Comment (metadave) [inline bin/irc-notify.sh]: yeah, I remembered seeing this file before.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          sh "bin/irc-notify.sh --stage 'build " + env.BRANCH_NAME + "' --status 'failed'"
+        throw err
+      }
+    }
+}
+
+def syncS3(String bucket) {
+    stage ('s3 sync') {
+        try {
+          sh "cd release && aws s3 sync . s3://" + bucket +" --acl public-read --delete --profile irl --exclude 'docs/*'"
```
Comment (jgmize) [inline Jenkinsfile]: Note the `--exclude docs`: the dir that you've built locally won't affect stage or prod, only the github pages which are optional now.

**Pull request merged**